src/os/bluestore/BlueStore.cc

```diff
@@ -2801,6 +2803,10 @@ void BlueStore::handle_conf_change(const struct md_config_t *conf,
       changed.count("bluestore_compression_max_blob_size")) {
     _set_compression();
   }
+  if (changed.count("bluestore_prefer_wal_size") ||
+      changed.count("bluestore_max_alloc_size")) {
+    _set_alloc_sizes();
```

Comment:
There doesn't appear to be any check to make sure that handle_conf_change doesn't happen before BlueStore::mount() (or mkfs/fsck, ie anything that opens _open_bdev).  As a result, bdev can be empty when _set_alloc_sizes() is called.